include/mfmg/dealii/belos_traits.hpp

```diff
@@ -0,0 +1,236 @@
+/*************************************************************************
+ * Copyright (c) 2017-2019 by the mfmg authors                           *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * This file is part of the mfmg libary. mfmg is distributed under a BSD *
+ * 3-clause license. For the licensing terms see the LICENSE file in the *
+ * top-level directory                                                   *
+ *                                                                       *
+ * SPDX-License-Identifier: BSD-3-Clause                                 *
+ *************************************************************************/
+
+#ifndef MFMG_BELOS_TRAITS_HPP
+#define MFMG_BELOS_TRAITS_HPP
+
+#include <mfmg/common/exceptions.hpp>
+#include <mfmg/dealii/multivector.hpp>
+
+#include <Teuchos_RCP.hpp>
+
+#include <BelosMultiVecTraits.hpp>
+
+#include <vector>
+
+// FIXME: Belos traits are only used for compilation
+// In the runtime, Belos is only used in TraceMin and KrylovSchur algorithms,
+// and TsqrOrthoManager
+
+namespace Belos
+{
+using mfmg::NotImplementedExc;
+
+template <typename VectorType>
+class MultiVecTraits<double, mfmg::MultiVector<VectorType>>
+{
+  using MultiVectorType = mfmg::MultiVector<VectorType>;
+
+public:
+  static Teuchos::RCP<MultiVectorType> Clone(const MultiVectorType &mv,
+                                             const int numvecs)
+  {
+    std::ignore = mv;
+    std::ignore = numvecs;
+    throw NotImplementedExc();
```

Comment:
You could use `ASSERT_NOT_IMPLEMENTED()` instead of throwing yourself.

Reply:
I thought we had that discussion...

> Actually, the sad part is that to use ASSERT_THROW_NOT_IMPLEMENTED() I will have to pair it with return statement, as it is implemented as a function in mfmg... which would also mean that I'll have to create a dummy object of the right type.